### PR TITLE
Fix minor flaws in receiver amounts calculation & add tests

### PR DESCRIPTION
**Pull request by stejbac**
Change the receiver amounts calculation in `bisq.trade.mu_sig.DelayedPayoutTxReceiverService` slightly, to fix a couple of minor flaws discovered while porting the algorithm to Rust (in order to calculate precise fees for the Redirect Tx in the original end-to-end protocol implementation, as part of the not-yet-merged https://github.com/bisq-network/bisq-musig/pull/94):

1. The intermediate floating-point receiver amounts should be rounded to nearest by `Math::round`, instead of down by casting to long, to eliminate bias, so that cumulative rounding errors grow at the square root of the output number, not linearly;

2. The fee-dependent `minOutputAmount` used in the algorithm should be rounded _up_ to the next whole satoshi value, not down though integer division by 1000.

Applying (1) potentially introduces a new problem, that in some edge cases too many outputs get included to pay each receiver the minimum allowed amount (say, if all the BM have equal shares paying just below the minimum, which then get intermediately rounded up to the minimum, causing all to get included and leading to total overpayment). So add some logic to prevent that.

Also, backport unit tests written for the Rust version which exercise those three issues in particular, and include a test for the associated helper method, `DelayedPayoutTxReceiverService::scriptPubKeyLength`.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved rounding for payout calculations to avoid tiny-sum errors
  * Stronger checks and early-stopping to prevent economically infeasible or undersized payouts

* **Tests**
  * Added comprehensive tests for payout distribution across many scenarios and address/script types, including edge-case rounding and large-output totals

### CODE REVIEW

File: trade/src/test/java/bisq/trade/mu_sig/DelayedPayoutTxReceiverServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        var burningmanDataList = burningmanDataList(ImmutableMap.of(
+                "bcrt1p88h9s6lq8jw3ehdlljp7sa85kwpp9lvyrl077twvjnackk4lxt0sffnlrk", 0.5,
+                "bcrt1phhl8d90r9haqwtvw2cv4ryjl8tlnqrv48nhpy7yyks5du6mr66xq5nlwhz", 0.5
+        ).entrySet());
+
+        // Computing the receiver list fails, as none could be economically included.
+        assertThrows(IllegalArgumentException.class, () ->
+                DelayedPayoutTxReceiverService.computeReceivers(
+                        burningmanDataList, availableAmountMsat, feeRateSatPerKwu));
+    }
```
[the remark this answers is not in the document]
Comment (stejbac) [reply trade/src/test/java/bisq/trade/mu_sig/DelayedPayoutTxReceiverServiceTest.java:144]: Yes, `availableAmountMsat` should be 1_429_000, not 1_149_008. (The wrong value was copied from the Rust tests.)

**Review by HenrikJannsen — APPROVED** [review]
utACK
